inst/shiny/modules/tab_tlg.R

```diff
@@ -268,19 +265,45 @@ tab_tlg_server <- function(id, data) {
                            choices = "")
       }
 
-      if (sum(tlg_order_filt$Type == "Graph") > 0) {
-        updateRadioButtons(
-          session = session,
-          inputId = "buttons_Graphs",
-          label = "Graph to display",
-          choices = tlg_order_filt$Label[tlg_order_filt$Type == "Graph"]
-        )
-      } else {
-        updateRadioButtons(session = session,
-                           inputId = "buttons_Graphs",
-                           label = "",
-                           choices = "")
-      }
+      tlg_order_graphs <- filter(tlg_order_filt, Type == "Graph") %>%
+        select("id") %>%
+        pull()
+
+      panels <- lapply(tlg_order_graphs, function(g_id) {
+        plot_ui <- {
+          g_def <- .TLG_DEFINITIONS[[g_id]]
+          module_id <- paste0(g_id, stringi::stri_rand_strings(1, 5))
+
+          if (exists(g_def$fun)) {
+            tlg_plot_server(module_id, get(g_def$fun), g_def$options)
+            tlg_plot_ui(session$ns(module_id))
+          } else {
+            tags$div("Plot not implemented yet")
+          }
+        }
+
+        nav_panel(g_def$label, plot_ui)
+      })
+
+      panels$"widths" <- c(2, 10)
+      output$graphs <- renderUI({
```

Comment:
suggestion: i'm strongly against having a render* function inside observe* function, because both are essentiall equally-powered observers.

instead, i suggest extracting `tlg_order_filt <- tlg_order()[tlg_order()$Selection, ]` into a separate reactive, and separate choice update and graphs rendering.